pkg/report/model/cyclonedx.go

```diff
@@ -0,0 +1,257 @@
+	ID              string           `xml:"v:id"`
+	Source          Source           `xml:"v:source"`
+	Ratings         []Rating         `xml:"v:ratings>v:rating"`
+	Description     string           `xml:"v:description"`
+	Recommendations []Recommendation `xml:"v:recommendations>v:recommendation"`
+}
+
+// Source includes information about the origin where the vulnerability was reported
+type Source struct {
+	Name string `xml:"name"`
+	URL  string `xml:"url"`
+}
+
+// Rating includes the severity of the vulnerability and the method used to classify it
+type Rating struct {
+	Severity string `xml:"v:severity"`
+	Method   string `xml:"v:method"`
+}
+
+// Recommendation describes how the vulnerability should be avoided
+type Recommendation struct {
+	Recommendation string `xml:""`
+}
+
+func match(fileNames []model.VulnerableFile, fileName string) bool {
+	for i := range fileNames {
+		if fileNames[i].FileName == fileName {
+			return true
+		}
+	}
+	return false
+}
+
+func getAllFiles(summary *model.Summary) []model.VulnerableFile {
+	var fileNames []model.VulnerableFile
+	for i := range summary.Queries {
+		files := summary.Queries[i].Files
+		for idx := range files {
+			if !match(fileNames, files[idx].FileName) {
+				fileNames = append(fileNames, files[idx])
+			}
+		}
+	}
+	return fileNames
+}
+
+func generateSha256(filePath string) string {
+	content, err := os.ReadFile(filepath.Clean(filePath))
+
+	if err != nil {
+		log.Trace().Msgf("failed to read %s", filePath)
+		return ""
+	}
+
+	hashSum := sha256.Sum256(content)
+	return hex.EncodeToString(hashSum[:])
+}
+
+func getPurl(filePath, version string) string {
+	return fmt.Sprintf("pkg:generic/%s@%s", filePath, version)
+}
+
+func getDescription(query *model.QueryResult) string {
+	queryDescription := query.Description
+
+	if query.CISDescriptionTextFormatted != "" {
+		queryDescription = query.CISDescriptionTextFormatted
+	}
+
+	description := fmt.Sprintf("[%s].[%s]: %s", query.Platform, query.QueryName, queryDescription)
+
+	return description
+}
+
+func getVulnerabilities(fileName, purl string, summary *model.Summary) []Vulnerability {
+	var vulns []Vulnerability
+	for i := range summary.Queries {
+		query := summary.Queries[i]
+		for idx := range query.Files {
+			file := query.Files[idx]
+			if fileName == file.FileName {
+				vuln := Vulnerability{
+					Ref: purl + query.QueryID,
+					ID:  query.QueryID,
+					Source: Source{
+						Name: "KICS",
+						URL:  "https://kics.io/",
+					},
+					Ratings: []Rating{
+						{
+							Severity: cycloneDxSeverityLevelEquivalence[query.Severity],
+							Method:   "Other",
+						},
+					},
+					Description: getDescription(&query),
+					Recommendations: []Recommendation{
+						{
+							Recommendation: fmt.Sprintf("In line %d, a result was found. '%s', but '%s'",
+								file.Line, file.KeyActualValue, file.KeyExpectedValue),
+						},
+					},
+				}
+				vulns = append(vulns, vuln)
+			}
+		}
+	}
+	return vulns
```

Comment:
I think this function could be refactored for readability, can you place from lines 163 to 189 in a separate function?